lib/Doctrine/ORM/EntityManager.php

```diff
@@ -382,8 +372,13 @@ public function find($entityName, $id, $lockMode = null, $lockVersion = null)
         }
 
         foreach ($id as $i => $value) {
-            if (is_object($value) && $this->metadataFactory->hasMetadataFor(StaticClassNameConverter::getClass($value))) {
-                $id[$i] = $this->unitOfWork->getSingleIdentifierValue($value);
+            if (is_object($value)) {
+                try{
+                    $this->metadatas->get(StaticClassNameConverter::getClass($value));
```

Comment:
Same here: move `StaticClassNameConverter` into a detail of `MetadataCollection`